src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -6860,17 +6859,17 @@ private bool TryReadSqlDateTime(SqlBuffer value, byte tdsType, int length, byte
 
                 case TdsEnums.SQLTIME:
                     Debug.Assert(3 <= length && length <= 5, "invalid length for time type!");
-                    value.SetToTime(datetimeBuffer, length, scale, scale);
+                    value.SetToTime(datetimeBuffer, scale, scale);
```

Comment:
The correct buffer size should be used after removing the length parameter.
> datetimeBuffer.Slice(0, length)

Reply:
Good catch. Updated to match the netcore version exactly.